Description: Disable bluetooth applet hack
 The hack prevent the Debian package to build properly. Modify it into a
 simple rpath addition in libcinnamon.so.
Author: Nicolas Bourdaud <nicolas.bourdaud@gmail.com>
Origin: vendor
Forwarded: https://github.com/linuxmint/Cinnamon/pull/987
Applied-Upstream: no
Last-Update: 2012-12-10

--- a/src/cinnamon-plugin.c
+++ b/src/cinnamon-plugin.c
@@ -382,17 +382,3 @@
 
   return &info;
 }
-
-#if HAVE_BLUETOOTH
-/* HACK:
-   Add a non-static function that calls into libgnome-bluetooth-applet.so,
-   to avoid the linker being too smart and removing the dependency.
-   This function is never actually called.
-*/
-extern GType bluetooth_applet_get_type(void);
-void _cinnamon_link_to_bluetooth(void);
-
-void _cinnamon_link_to_bluetooth(void) {
-  bluetooth_applet_get_type();
-}
-#endif
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -263,10 +263,12 @@
 EXTRA_DIST += cinnamon-enum-types.c.in
 
 libcinnamon_la_LDFLAGS = -avoid-version
+if BUILD_WITH_BLUETOOTH
+libcinnamon_la_LDFLAGS += -Wl,-rpath=$(BLUETOOTH_DIR)
+endif
 libcinnamon_la_LIBADD =		\
 	-lm			\
 	$(CINNAMON_LIBS)	\
-	$(BLUETOOTH_LIBS)	\
 	libst-1.0.la       	\
 	libtray.la		\
 	libgvc.la		\
--- a/configure.ac
+++ b/configure.ac
@@ -121,10 +121,12 @@
 	 AC_DEFINE_UNQUOTED([BLUETOOTH_DIR],["$BLUETOOTH_DIR"],[Path to installed GnomeBluetooth typelib and library])
 	 AC_DEFINE([HAVE_BLUETOOTH],[1],[Define if you have libgnome-bluetooth-applet])
 	 AC_SUBST([HAVE_BLUETOOTH],[1])
+	 build_with_bluetooth=true;
 	 AC_MSG_RESULT([yes])],
 	[AC_DEFINE([HAVE_BLUETOOTH],[0])
 	 AC_SUBST([HAVE_BLUETOOTH],[0])
 	 AC_MSG_RESULT([no])])
+AM_CONDITIONAL([BUILD_WITH_BLUETOOTH], [test "x$build_with_bluetooth" = xtrue])
 
 MUFFIN_GIR_DIR=`$PKG_CONFIG --variable=girdir libmuffin`
 MUFFIN_TYPELIB_DIR=`$PKG_CONFIG --variable=typelibdir libmuffin`
